Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,332 @@
+          ? RebarStyle.Standard
+          : RebarStyle.StirrupTie;
 
-      // find some way to test if curves should be converted to curve loops for freeform rebar or not for rebar from curves
-      var curveLoops = closedCurves.Select(o => CurveArrayToCurveLoop(o)).ToList();
-
-      DB.Structure.Rebar rebar = null;
+      // get start and end hooks and orientations
+      var speckleStartHook = speckleRebar.startHook as RevitRebarHook;
+      RebarHookType startHook = null;
+      RebarHookOrientation startHookOrientation = RebarHookOrientation.Right;
+      if (speckleStartHook != null)
+      {
+        startHook = RebarHookToNative(speckleStartHook);
+        Enum.TryParse(speckleStartHook.orientation, out startHookOrientation);
+      }
+      var speckleEndHook = speckleRebar.endHook as RevitRebarHook;
+      RebarHookType endHook = null;
+      RebarHookOrientation endHookOrientation = RebarHookOrientation.Right;
+      if (speckleEndHook != null)
+      {
+        endHook = RebarHookToNative(speckleEndHook);
+        Enum.TryParse(speckleEndHook.orientation, out endHookOrientation);
+      }
 
-      // get host element
-      var host = GetExistingElementByApplicationId(speckleRevitRebar.host);
-      if (host == null)
+      // get the shape curves
+      List<DB.Curve> curves = barShape.curves.SelectMany(o => CurveToNative(o).Cast<DB.Curve>()).ToList();
+      if (curves is null || !curves.Any())
       {
-        appObj.Update(status: ApplicationObject.State.Failed, logItem: "Rebar host not found.");
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: "Could not convert any shape curves");
         return appObj;
       }
 
-      if (docObj != null)
+      // get the rebar plane norm from the curves
+      XYZ normal = XYZ.BasisZ;
+      if (speckleRebar.normal is not null)
       {
-        rebar = (DB.Structure.Rebar)docObj;
-        // if the number of curves don't match, we need to create a new bar
-        if (rebar.Quantity != closedCurves.Count + openCurves.Count)
-          rebar = null;
+        normal = VectorToNative(speckleRebar.normal);
       }
 
-      // update curves if we can, only available in revit 2022
-#if REVIT2022
-      if (rebar != null)
-        rebar.MoveBarInSet(0, DB.Transform.Identity);
-#endif
-
-      // create freeform from curveloops if they exist
-      if (curveLoops.Count > 0)
+      // create the rebar
+      DB.Structure.Rebar rebar = null;
+      try
       {
-        var result = RebarFreeFormValidationResult.Success;
-        rebar ??= DB.Structure.Rebar.CreateFreeForm(Doc, barType, host, curveLoops, out result);
-        if (result != RebarFreeFormValidationResult.Success)
-        {
-          appObj.Update(status: ApplicationObject.State.Failed, logItem: "Freeform Rebar could not be created from closed curves.");
-          return appObj;
-        }
+        rebar = DB.Structure.Rebar.CreateFromCurves(
```

Comment:
Ha, sorry, I spent 5 minutes considering this as being quite a long signature and then realised this is the Revit AP! Nice going Revit! :)